galaxy_importer/loaders.py

```diff
@@ -134,29 +170,30 @@ def handle_nested_tables(obj, table_key):
                 for row in obj[table_key]:
                     handle_nested_tables(row, table_key)
 
-        doc = data[self.name].get('doc')
+        doc = data.get('doc', None)
         if doc and 'options' in doc.keys() and isinstance(doc['options'], dict):
             doc['options'] = dict_to_named_list(doc['options'])
             for d in doc['options']:
                 handle_nested_tables(d, table_key='suboptions')
 
-        ret = data[self.name].get('return')
+        ret = data.get('return', None)
         if ret and isinstance(ret, dict):
-            data[self.name]['return'] = dict_to_named_list(ret)
-            for d in data[self.name]['return']:
+            data['return'] = dict_to_named_list(ret)
+            for d in data['return']:
                 handle_nested_tables(d, table_key='contains')
 
         return data
 
     def _run_ansible_doc(self):
         cmd = [
+            'env', f'ANSIBLE_COLLECTIONS_PATHS={self.tmp_dir}',
```

Comment:
Since the ansible-doc output has a 100% chance of changing randomly in the future, it may be useful to extract the ansible-doc and doc strings handling code to it's own module or class if it makes it easier to test and change.

Reply:
test_loader_doc_strings.py being a separate module also seems to support that idea.

Reply:
agreed and incorporated